### PR TITLE
[Feature] Implement CSV export options

### PR DESCRIPTION
**Pull request by willumz**
Satisfy part of #34 by adding a form to allow exporting dice roll data to CSV files for external analysis.

### CODE REVIEW

**Comment by willumz** [comment]
Hi! I found myself in need of this feature and so thought I'd try tackling it myself. Obviously I'm not extremely familiar with the codebase or the fine workings so let me know if there's anything that needs changing and so on.

**Comment by jacobwojoski** [comment]
Looks Cool! I’ll take a look and do some testing after work today

**Comment by willumz** [comment]
Hi, sorry, I don't think I got pinged when you edited your message. I'm struggling to spot your comments in GitHub - would you be able to note the file/line numbers for me?

**Comment by jacobwojoski** [comment]
All good. It wasn't showing the review comment  unless i signed in so might be a perms issue or something. Here's a screenshot of the comments. They were both minor and up to how you want it implemented  for now

image

image

**Comment by jacobwojoski** [comment]
You might also be able to see the comments if you open the files changed page of this PR

File: scripts/classes/forms/dice-stats-exportdatapage.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    //Handle button events made on the form
+    async _handleButtonClick(event){
+        const clickedElement = $(event.currentTarget);
+        const action = clickedElement.data().action;
+
+        if(DS_GLOBALS.FORM_EXPORT == null){return;}
+
+        if (!game.user.isGM) return;
+
+        switch(action) {
```
[the remark this answers is not in the document]
Comment (willumz) [reply scripts/classes/forms/dice-stats-exportdatapage.js:70]: When you say a single file with a line offset, would you be able to give an example? I've never formatted a CSV with multiple tables in it before - or do you mean something else and I misunderstood?

[the remark this answers is not in the document]
Comment (jacobwojoski) [reply scripts/classes/forms/dice-stats-exportdatapage.js:70]: You could split up the csv rows by using newline characters. `\n`
Now you can read in all the data from one file this way.

D2 info info on line X, D3=X+1, D4=X+2 etc where X might be 0 or 1 if you have a heading row for the CSV

csv would end up looking something like 
`0,2,\n2,2,3,\n6,6,6,6\n`

I cant remember if the newline needs to be in its own space or not though. 
EX: ` ,\n<next_row_value>  vs  \n,<next_row_value>`

[the remark this answers is not in the document]
Comment (willumz) [reply scripts/classes/forms/dice-stats-exportdatapage.js:70]: Ah, I see what you mean - at the moment rows represent the different players, which is why I did separate CSVs (there are three dimensions to the data here). Could have an entry for each player for each die type
i.e.
```csv
Username,1,2,3,4,5,6,7,8,...
Player1 (D4),...
Player2 (D4),...
Player1 (D6),...
Player2 (D6),...
...
```
I felt just having multiple tables for each would be better but can change it if you like.
I agree the UI looks very crowded with all the buttons for each die type - maybe a dropdown to select which you'd like would be better?

[the remark this answers is not in the document]
Comment (willumz) [reply scripts/classes/forms/dice-stats-exportdatapage.js:70]: Or perhaps it should be the other way around? Does should represent each type of die and you should download a different file for each player - that would probably be better, then there could be a combined CSV as well which just summed all the players together for an overall view

[the remark this answers is not in the document]
Comment (jacobwojoski) [reply scripts/classes/forms/dice-stats-exportdatapage.js:70]: Ah right forgot about that. 
Id say different file for each player and combined file sounds good for now. could away just add a dieType collumn to the csv too. but thats also a bit annoying. however you feel is best for your use case works for now

**Comment by willumz** [comment]
Okay, those changes have been made, thanks for your help 👍🏻

**Pull request merged**